presto-hive/src/main/java/io/prestosql/plugin/hive/security/SqlStandardAccessControl.java

```diff
@@ -433,11 +435,15 @@ private boolean checkTablePermission(
         }
 
         SemiTransactionalHiveMetastore metastore = metastoreProvider.apply(((HiveTransactionHandle) context.getTransactionHandle()));
-        return metastore.listTablePrivileges(new HiveIdentity(context.getIdentity()), tableName.getSchemaName(), tableName.getTableName(), Optional.empty())
-                .stream()
-                .filter(privilegeInfo -> listEnabledPrincipals(metastore, context.getIdentity()).anyMatch(privilegeInfo.getGrantee()::equals))
+
+        Set<HivePrincipal> allowedPrincipals = metastore.listTablePrivileges(new HiveIdentity(context.getIdentity()), tableName.getSchemaName(), tableName.getTableName(), Optional.empty()).stream()
```

Comment:
Include rationale in `Do not enumerate all enabled roles when checking access in sql-standard` commit